src/openvas.c

```diff
@@ -192,82 +195,245 @@ init_signal_handlers (void)
+                  free (buf);
+                  json_reader_end_element (reader);
+                }
+              g_debug ("%s: %s -> %s", __func__, key, values);
+              prefs_set (key, values);
+              free (values);
+            }
+        }
+      // dictionary (plugins)
+      // parse list semicolon separated into single string
+      if (json_reader_is_object (reader))
+        {
+          if (!strcmp (key, "plugins"))
+            {
+              const char *value;
+              char *values;
+              int len, j, num_plugins;
+
+              json_reader_read_member (reader, "single_vts");
+              num_plugins = json_reader_count_elements (reader);
+
+              key = "plugin_set";
+
+              if (num_plugins > 0)
+                {
+                  // Write first plugin oid
+                  json_reader_read_element (reader, 0);
+                  json_reader_read_member (reader, "oid");
+                  value = json_reader_get_string_value (reader);
+                  len = strlen (value);
+                  values = (char *) (malloc (sizeof (char) * len + 1));
+                  snprintf (values, len + 1, "%s", value);
+                  json_reader_end_member (reader);
+                  json_reader_end_element (reader);
+
+                  // Append all otherplugins semicolon separated
+                  for (j = 1; j < num_plugins; j++)
+                    {
+                      json_reader_read_element (reader, j);
+                      json_reader_read_member (reader, "oid");
+                      value = json_reader_get_string_value (reader);
+                      len += strlen (value);
+                      char *buf = values;
+                      values = (char *) (malloc (sizeof (char) * len + 1));
+                      snprintf (values, len + 1, "%s;%s", buf, value);
+                      free (buf);
+                      json_reader_end_member (reader);
+                      json_reader_end_element (reader);
+                    }
+                  g_debug ("%s: %s -> %s", __func__, key, values);
+                  prefs_set (key, values);
+                  free (values);
+                }
+              json_reader_end_member (reader);
+            }
+        }
+      json_reader_end_member (reader);
+    }
 
-  kb_item_free (res);
+  g_object_unref (reader);
+  g_object_unref (parser);
+  g_free (members);
   return 0;
 }
 
+/**
+ * @brief Read the scan preferences from mqtt
+ *
+ * Adds preferences to the global_prefs.
+ * If preference already exists in global_prefs they will be overwritten by
+ * prefs from client.
+ *
+ * @param globals Scan ID of globals used as key to find the corresponding KB
+ * where to take the preferences from.
+ *
+ * @return 0 on success, -1 if the kb is not found or no prefs are found in
+ *         the kb.
+ */
+static int
+overwrite_openvas_prefs_with_prefs_from_client (struct scan_globals *globals)
```

Comment:
It would be nice if the original function could not be touched at all at this point.
Just put the functionality in a new function and call that one instead of `overwrite_openvas_prefs_with_prefs_from_client()`.
I know it means dead code but I do think for now it would be nice to have it as reference for how openvas handles the prefs currently because there are still some unknowns here.
You can use `__attribute__ ((unused))` if the compiler complains. 

Alternatively this function could be just touched minimally with `if/else mqtt_is_initialized()` eulabeia switch.

Reply:
Wouldn't it be better to run in those unknown cases explicitly to make them known instead of keeping it around just in case?

Reply:
Maybe I was not really clear what I meant. The cases are known.

For example before the changes there is the logic which at the end does `store_file()`. But this PR just deletes this functionality without replacement and without documenting that the functionality was now removed. So the information about what we still need to implement in the future gets lost on code side.
For our start scan example this PR is enough. But we still need to handle certificates and so on which are provided as file and stored as such (`store_file`) at the end. I thought handling it the way I proposed may be the easiest and fastest hacky solution for now so that functionality which we still need at the end is not getting lost in the void.

Reply:
Maybe? Since this pr is based on another pr and I don't know the internal certificate handling I'm still not 100% sure what you have in mind. Probably it makes sense that you could add a missing comment?